portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+ // generate the project contents when the project.html document is loaded
+window.onload = function generateProjectContent() {
```

Comment:
What are the pros and cons of generating the content in code, instead of serving it directly in HTML?

Reply:
I was able to skip rewriting tags for multiple times, but it seems like the page loads slower now.

Reply:
Overall, prefer to use static html when possible and only resort to javascript when needed.

HTML is going to be more performant for the end user and use less processing power from their browser. Javascript is useful for making things interactive, but shouldn't be used for static content.

Reply:
Makes sense. I changed it back on html :)